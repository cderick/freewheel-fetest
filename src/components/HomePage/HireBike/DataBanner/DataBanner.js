import React from 'react'
import PropTypes from 'prop-types'
import s from './DataBanner.module.scss'

class DataBanner extends React.Component {
	render() {
		const { cardsData, queryValue, showContent } = this.props
		if (!cardsData) return null
		return (
			<div id="DataBanner" className="container pt-5 pb-5">
				{showContent &&
				<section>
					<div className="row">
						<div className="col text-center">
							<h2 className="pb-5">You searching at: <span className="text-danger">{queryValue}</span></h2>
						</div>
					</div>
					<div className="row">
						<div className="col">
							{cardsData && cardsData.length
								&& cardsData.map((cv, ind) => (
									<div key={`map${ind}`} className={`card float-left mr-3 mb-3 ${s.customWidth}`}>
										<img src="https://wallpapercave.com/wp/66stx2A.jpg" className="card-img-top" alt="catPlaceholder" />
										<div className="card-body">
											<h6 className="card-title text-danger font-weight-bold">{cv.commonName}</h6>
											<p className="card-text">{`${(cv.id).substring(((cv.id).indexOf('_') + 1))} ${cv.commonName} (${cv.lat}, ${cv.lon})`}</p>
											<a href={`https://www.google.com/maps/search/${cv.lat},${cv.lon}/${cv.lat},${cv.lon},17z`} rel="noopener noreferrer" target="_blank" className="btn btn-primary">Check on Google Maps</a>
										</div>
									</div>
								))}
						</div>
					</div>
				</section>}
			</div>
		)
	}
}

DataBanner.propTypes = {
	cardsData: PropTypes.arrayOf(PropTypes.any),
	queryValue: PropTypes.string,
	showContent: PropTypes.bool
}

DataBanner.defaultProps = {
	cardsData: null,
	queryValue: undefined,
	showContent: false
}

export default DataBanner
