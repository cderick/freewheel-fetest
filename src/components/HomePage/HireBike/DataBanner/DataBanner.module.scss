.customWidth {
  width: 23%;
  height: 360px;
}
