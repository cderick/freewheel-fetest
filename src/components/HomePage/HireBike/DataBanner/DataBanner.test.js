import React from 'react'
import renderer from 'react-test-renderer'
import DataBanner from './DataBanner'

const minProps = {
	cardsData: {},
    queryValue: 'Waterloo'
}

const componentNoprops = renderer.create(<DataBanner />)
const componentWithProps = renderer.create(<DataBanner {...minProps}  />)

describe('Test on <DataBanner /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
   test('should render with min props', () => {
      const treeProps = componentWithProps.toJSON()
      expect(treeProps).toMatchSnapshot()
   })
})