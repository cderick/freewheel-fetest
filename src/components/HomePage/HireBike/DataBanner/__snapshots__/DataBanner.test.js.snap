// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <DataBanner /> component should not render with no props 1`] = `null`;

exports[`Test on <DataBanner /> component should render with min props 1`] = `
<div
  className="container pt-5 pb-5"
  id="DataBanner"
/>
`;
